test/e2e/workflow/deployment_test.go

```diff
@@ -281,6 +282,34 @@ var _ = Describe("NetworkAddonsConfig", func() {
 			CreateConfig(gvk, configSpec)
 			CheckConfigCondition(gvk, ConditionAvailable, ConditionTrue, 15*time.Minute, 30*time.Second)
 		})
+		Context("and checking CNAO prometheus endpoint", func() {
+			var err error
+			var scrapedData string
+			BeforeEach(func() {
+				scrapedData, err = GetScrapedDataFromMonitoringEndpoint()
+				Expect(err).ToNot(HaveOccurred())
+			})
+			It("Should report the expected metrics", func() {
+				expectedmetricValueMap := map[string]string{
+					"kubevirt_cnao_ready": "1",
+				}
+
+				for metricName, expectedValue := range expectedmetricValueMap {
+					metricFound := false
+					for _, line := range strings.Split(scrapedData, "\n") {
+						if strings.HasPrefix(line, "#") {
+							continue
+						}
+						if strings.Contains(line, metricName) {
```

Comment:
its problematic in the rare case where there is metric name which is part of other metric name

Reply:
I agree. took your suggestion with slight changes.